src/esp/io/URDFParser.h

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Code adapted from Bullet3/examples/Importers/ImportURDFDemo ...
+
+#pragma once
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Color.h>
+#include <Magnum/Math/Matrix4.h>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace tinyxml2 {
+class XMLElement;
+};
+
+////////////////////////////////////
+// Utility/storage structs
+////////////////////////////////////
+
+namespace esp {
+namespace io {
+
+struct UrdfMaterialColor {
+  Magnum::Color4 m_rgbaColor;
+  Magnum::Color3 m_specularColor;
+  UrdfMaterialColor()
+      : m_rgbaColor(0.8, 0.8, 0.8, 1), m_specularColor(0.4, 0.4, 0.4) {}
```

Comment:
I don't have much context with URDF so I'm phrasing this as a question: Is an unspecified material or color very common, or is it more of an error case?

If it's more of an error case, it might be better to use an obnoxious color like (1, 0, 1) magenta, so when users complain about their model being pink, we know right away the problem.